### PR TITLE
Add exchange for union with TableScan and Values

### PR DESCRIPTION
**Pull request by sopel39**
Extracted-From: https://github.com/prestodb/presto/pull/11957

### CODE REVIEW

File: presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestAddExchangesPlans.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.join;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.tableScan;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.values;
+import static io.prestosql.sql.planner.plan.ExchangeNode.Scope.REMOTE;
+import static io.prestosql.sql.planner.plan.ExchangeNode.Type.REPARTITION;
+import static io.prestosql.sql.planner.plan.JoinNode.Type.INNER;
+
+/**
+ * These are plan tests similar to what we have for other optimizers (e.g. {@link io.prestosql.sql.planner.TestPredicatePushdown})
+ * They test that the plan for a query after the optimizer runs is as expected.
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestAddExchangesPlans.java]: The above 2 lines are useless, i would remove the whole javadoc

File: presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestAddExchangesPlans.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+ * These are separate from {@link TestAddExchanges} because those are unit tests for
+ * how layouts get chosen.
+ */
+public class TestAddExchangesPlans
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestAddExchangesPlans.java]: Why is it separate from `TestLogicalPlanner`?

Comment (sopel39) [reply presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestAddExchangesPlans.java]: I think they are mean to test that specific code in `AddExchanges` triggers, which is fine by me.

**Comment by sopel39** [comment]
I've reviewed the original code, landing

**Pull request merged**